karabo_bridge/client.py

```diff
@@ -87,32 +100,38 @@ def next(self):
 
     def _deserialize(self, msg):
         if len(msg) < 2:
-            return self._deserializer(msg[-1])
-
-        dat = {}
-        for header, data in zip(*[iter(msg)]*2):
+            data = self._deserializer(msg[-1])
+            meta = {}
+            for key, value in data.items():
+                meta[key] = value.get('metadata', {})
+            return data, meta
+
+        data = {}
+        meta = {}
+        for header, payload in zip(*[iter(msg)]*2):
             md = self._deserializer(header)
             source = md['source']
             content = md['content']
+            meta[source] = md.get('metadata', {})
 
             if content in ('msgpack', 'pickle.HIGHEST_PROTOCOL',
                            'pickle.DEFAULT_PROTOCOL'):
-                dat[source] = self._deserializer(data)
+                data[source] = self._deserializer(payload)
             elif content in ('array', 'ImageData'):
```

Comment:
Isn't `ImageData` gone as a separate content type?

Reply:
it is, but older version of the server still use this. I would remove it when we are sure that we won't use older version anymore, i.e. after this run cycle probably.